### PR TITLE
Add highest property to RoleStore

### PR DESCRIPTION
**Pull request by Spindlyskit**
**Please describe the changes this PR makes and why it should be merged:**
This commit adds a higest property to the RoleStore DataStore. This just makes it easier to get the highest role on a guild and makes the RoleStore class feel more like the GuildMemberRoleStore class. This PR should be merged because it adds a small utility method that would be helpful to have.

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/stores/RoleStore.js [hunk 1]
```diff
@@ -55,6 +55,15 @@ class RoleStore extends DataStore {
     });
   }
 
+  /**
+   * The role with the highest position in the store
+   * @type {Role}
+   * @readonly
+   */
+  get highest() {
+    return this.reduce((prev, role) => !prev || role.comparePositionTo(prev) > 0 ? role : prev);
```
Comment (kyranet) [inline src/stores/RoleStore.js]: Another option for that would be:
```javascript
return this.reduce((prev, role) => role.comparePositionTo(prev) > 0 ? role : prev, this.first());
```

Which sets the first role (normally the everyone role) as the first `prev` value. That'd skip one more check and improve performance and efficiency when checking for large `RoleStore`s. Remember that [`Collection#reduce`](https://discord.js.org/#/docs/main/master/class/Collection?scrollTo=reduce) is identical to [`Array#reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce), so you can take advantage of this.

**Comment by Spindlyskit** [comment]
the current code is almost completely copied from GuildMemberRoleStore so would it be worth changing that as well?

**Comment by kyranet** [comment]
Worth a try, I guess.

**Comment by Lewdcario** [comment]
@Spindlyskit update on this PR?

**Pull request closed without merging**